VultisigApp/VultisigApp/Views/Vault/VaultPairDetailView.swift

```diff
@@ -38,16 +37,48 @@ struct VaultPairDetailView: View {
                     .frame(maxWidth: .infinity, alignment: .center)
                 
                 ForEach(devicesInfo, id: \.Index) { device in
-                    if device.Signer == vault.localPartyID {
-                        VaultPairDetailCell(title: device.Signer + " (This device)", description: .empty)
-                    } else {
-                        VaultPairDetailCell(title: device.Signer, description: .empty)
-                    }
+                    getDeviceCell(for: device)
                 }
             }
             .padding(.top, 30)
         }
     }
+    
+    var titleCell: some View {
+        Text(titleText())
+            .foregroundColor(.neutral0)
+            .font(.body20MenloBold)
+            .padding(.vertical, 22)
+            .padding(.horizontal, 16)
+            .frame(maxWidth: .infinity, alignment: .leading)
+            .background(Color.blue600)
+            .cornerRadius(10)
+            .padding(.horizontal, 16)
+            .lineLimit(3)
+    }
+    
+    private func titleText() -> String {
+        let name = vault.name
+        let dash = " - "
+        let part = NSLocalizedString("part", comment: "")
+        let of = NSLocalizedString("of", comment: "")
+        let space = " "
+        let vaultIndex = "\(vault.getThreshold() + 1)"
```

Comment:
<img width="830" alt="image" src="https://github.com/user-attachments/assets/3207822d-f921-4ce0-8ddb-488daa757cad">
I think here we should get the device's index + 1 , not `vault.getThreshold`